File: Strings/First_Non_Repeating_Character/firstNonRepeatingCharacter.js

function firstNonRepeatedCharacter(string) {
    const charCounter = {};
    for(let i = 0; i < string.length; i++) {
        const char = string.charAt(i);
        if(!(char in charCounter)) {
            charCounter[char] = 0;
        }
        charCounter[char]++;
    }
    for(const c in charCounter) {
        if(charCounter[c] === 1) {
            return c;
        }
    }
    return null;
}

console.log(firstNonRepeatedCharacter("abcdabcdef") === "e");
console.log(firstNonRepeatedCharacter("abcdacdef") === "b");
console.log(firstNonRepeatedCharacter("abcda") === "b");
console.log(firstNonRepeatedCharacter("abcd") === "a");

File: Strings/First_Non_Repeating_Character/first_non_repeating_character.py

def first_non_repeated_character(str):
    charCounter = {}
    for c in str:
        if c not in charCounter:
            charCounter[c] = 0
        charCounter[c] += 1
    for c in str:
        if charCounter[c] == 1:
            return c
    return None

print(first_non_repeated_character("abcdabcdef") == "e")
print(first_non_repeated_character("abcdacdef") == "b")
print(first_non_repeated_character("abcda") == "b")
print(first_non_repeated_character("abcd") == "a")

